eth/ethtest/utils_test.go

```diff
@@ -0,0 +1,300 @@
+
+	tosign := fmt.Sprintf("%s:%d", owner.String(), nonce)
+	msghash := crypto.Keccak256([]byte(tosign))
+	signed := validatorData.masterKey.Sign(string(msghash))
+	sig := signed.Serialize()
+
+	if !signed.VerifyByte(validatorData.masterPubKey, msghash) {
+		return nil, errors.New("couldn't sign correctly")
+	}
+
+	sharesData := append(pubkeys, encryptedShares...)
+	sharesDataSigned := append(sig, sharesData...)
+
+	return sharesDataSigned, nil
+}
+
+func setupEventHandler(
+	t *testing.T,
+	ctx context.Context,
+	logger *zap.Logger,
+	operator *testOperator,
+	ownerAddress *ethcommon.Address,
+	useMockCtrl bool,
+) (*eventhandler.EventHandler, *mocks.MockController, *gomock.Controller, operatorstorage.Storage, error) {
+	db, err := kv.NewInMemory(logger, basedb.Options{
+		Ctx: ctx,
+	})
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	storageMap := ibftstorage.NewStores()
+	nodeStorage, operatorData := setupOperatorStorage(logger, db, operator, ownerAddress)
+	testNetworkConfig := networkconfig.TestNetwork
+
+	keyManager, err := ekm.NewETHKeyManagerSigner(logger, db, testNetworkConfig, true, "")
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	ctrl := gomock.NewController(t)
+	bc := beacon.NewMockBeaconNode(ctrl)
+
+	contractFilterer, err := contract.NewContractFilterer(ethcommon.Address{}, nil)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	if useMockCtrl {
+		validatorCtrl := mocks.NewMockController(ctrl)
+
+		parser := eventparser.New(contractFilterer)
+
+		eh, err := eventhandler.New(
+			nodeStorage,
+			parser,
+			validatorCtrl,
+			testNetworkConfig.Domain,
+			validatorCtrl,
+			nodeStorage.GetPrivateKey,
+			keyManager,
+			bc,
+			storageMap,
+			eventhandler.WithFullNode(),
+			eventhandler.WithLogger(logger),
+		)
+
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
+		validatorCtrl.EXPECT().GetOperatorData().Return(operatorData).AnyTimes()
+
+		return eh, validatorCtrl, ctrl, nodeStorage, nil
+	}
+
+	validatorCtrl := validator.NewController(logger, validator.ControllerOptions{
+		Context:         ctx,
+		DB:              db,
+		RegistryStorage: nodeStorage,
+		KeyManager:      keyManager,
+		StorageMap:      storageMap,
+		OperatorData:    operatorData,
+	})
+
+	parser := eventparser.New(contractFilterer)
+
+	eh, err := eventhandler.New(
+		nodeStorage,
+		parser,
+		validatorCtrl,
+		testNetworkConfig.Domain,
+		validatorCtrl,
+		nodeStorage.GetPrivateKey,
+		keyManager,
+		bc,
+		storageMap,
+		eventhandler.WithFullNode(),
+		eventhandler.WithLogger(logger),
+	)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	return eh, nil, ctrl, nodeStorage, nil
+}
+
+func setupOperatorStorage(
+	logger *zap.Logger,
+	db basedb.Database,
+	operator *testOperator,
+	ownerAddress *ethcommon.Address,
+) (operatorstorage.Storage, *registrystorage.OperatorData) {
+	if operator == nil {
+		logger.Fatal("empty test operator was passed", zap.Error(fmt.Errorf("empty test operator was passed")))
```

Comment:
I think the error field is redundant here 

```suggestion
		logger.Fatal("empty test operator was passed")
```

Reply:
Fixed. Thanks